### PR TITLE
Add rotating rectangle as a drawing tool

### PR DESCRIPTION
**Pull request by CKrawczyk**
Staging branch URL: https://rotate-rect.pfe-preview.zooniverse.org/

Describe your changes.

This PR adds a rotating rectangle as a new drawing tool.

~~At the moment the drag handles do not work as expected when the rectangle is rotated. I think this has to do with the coordinate system the drag even is in being different than the one the `svg` object is in.~~

~~This is currently in coffee script since I started with the rectangle drawing tool's code.  I plan to convert it to ES6 once it is working.~~ The code is now in ES6.

Things left to do
- [x] fix drag handles when rectangle is rotated
- [x] convert to ES6

# Required Manual Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [x] Does talk load correctly?
- [x] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by CKrawczyk** [comment]
Everything is now working correctly for this PR.

**Comment by eatyourgreens** [comment]
Checking this out quickly, I find it weird that the rectangle rotates around the top left corner, rather than the centre. Maybe test this out with a few people to see if they feel the same?

**Comment by CKrawczyk** [comment]
I tried that at first and had a hard time getting the resize handles work nicely (the point of rotation is constantly changing as it is being resized).   I will give it another try.

**Comment by eatyourgreens** [comment]
Ask a couple of other people before you put a lot of effort into it. I might be the only person who finds this hard to use.

**Comment by CKrawczyk** [comment]
I have updated the code to have the rectangle rotate about the center.

To make the drag handles work nicely the code has been updated so the center of the rectangle stays fixed during resize.  The staging link has been updated.

File: app/classifier/drawing-tools/index.coffee [hunk 1]
```diff
@@ -14,4 +14,5 @@ module.exports =
   point: require './point'
   polygon: require './polygon'
   rectangle: require './rectangle'
+  rotateRectangle: require('./rotate-rectangle.jsx').default
```
Comment (eatyourgreens) [inline app/classifier/drawing-tools/index.coffee]: Doesn't this need the `.jsx` extension? We don't use it anywhere else that we require JSX components into CJSX files.

File: app/classifier/drawing-tools/rotate-rectangle.jsx [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  constructor(props) {
+    super(props);
+    this.getDeletePosition = this.getDeletePosition.bind(this);
+    this.handleMainDrag = this.handleMainDrag.bind(this);
+    this.handleTopLeftDrag = this.handleTopLeftDrag.bind(this);
+    this.handleTopRightDrag = this.handleTopRightDrag.bind(this);
+    this.handleBottomLeftDrag = this.handleBottomLeftDrag.bind(this);
+    this.handleBottomRightDrag = this.handleBottomRightDrag.bind(this);
+    this.handleRotate = this.handleRotate.bind(this);
+    this.normalizeMark = this.normalizeMark.bind(this);
```
Comment (eatyourgreens) [inline app/classifier/drawing-tools/rotate-rectangle.jsx]: I'm not sure, but I don't think you need to bind `normalizeMark` here. Instance methods are bound to an instance of a class by default in javascript. Only event handlers need to be explicitly bound.

File: app/classifier/drawing-tools/index.coffee (hunk 1 above)
Comment (CKrawczyk) [reply app/classifier/drawing-tools/index.coffee]: Good catch, this is a hold over from when I was converting out of `cjsx`, I will remove it.

Comment (eatyourgreens) [reply app/classifier/drawing-tools/index.coffee]: Cool. I'll hold off on merging until you've updated the code.

File: app/classifier/drawing-tools/rotate-rectangle.jsx (hunk 2 above)
Comment (eatyourgreens) [reply app/classifier/drawing-tools/rotate-rectangle.jsx]: I missed that it's also an event handler, for `onEnd`, which does change the `this` context.

**Pull request merged**